Actuator.m

```diff
@@ -198,8 +198,32 @@ function tune_pid_controller(obj, saturation, p_gain, i_gain, dv_gain, de_gain)
             configuration = [p_gain, i_gain, dv_gain, de_gain , saturationL_H];
             obj.write_multi_registers(133, 6, configuration);  % PC_PGAIN 133, num consecutive registers 6, register values  configuration
         end
+
         
+        %% Change Motor Direction
+        % Sets the motor direction to 0 or 1
+        function set_direction(obj, direction)
+            obj.write_register(152, direction);   %POS_SIGN
+        end
+
+        %% Configure AutoZero
+        % Set autozero with configuration:
+        %ZERO_MODE: 0: zero negative positions, 1: manual zeroing, 2: Auto Zero
+        %Enabled, 3:Auto Zero on Boot
+        %EXIT_MODE: 1: sleep, 2: force, 3: position,4: Haptic,5:Kinematic
+        function configure_zero(obj, ZERO_MODE, AUTO_ZERO_FORCE_N, AUTO_ZERO_EXIT_MODE)
+            configuration = [ZERO_MODE AUTO_ZERO_FORCE_N AUTO_ZERO_EXIT_MODE];
+            obj.write_multi_registers(171, 3, configuration);  %autozero configuration writing
+        end
 
+        %% Trigger Autozero
+        % Trigger Autozero using ctrl register
+        function trigger_zero(obj)
+            obj.write_register(3, 55);   %Trigger Zeroing
+            while obj.read_register(317,1)==55
```

Comment:
Is this meant to stay on this line until auto zeroing completes?
This isn't very evident from the function name. I would either rename it something like "auto zero blocking" or alternatively just write your program to wait for the mode to change.

Also, this is probably only working due to the slow speed of the Modbus comms. When you write CTRL_REG_3 with the mode, the MODE_OF_OPERATION register (317) will not be updated until that mode change command is parsed. If line 223 were evaluated immediately after sending the write register, the mode would probably still be whatever it was before.

Reply:
Good call, best method probably to set a 250ms sleep before reading the register?